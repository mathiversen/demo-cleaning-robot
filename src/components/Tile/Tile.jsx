import React from 'react'
import { css } from 'goober'

export default ({ x, y, clean, children }) => {
    const uniqueClass = css`
        grid-column: ${x + 1};
        grid-row: ${y + 1};
        background: ${clean ? '#00a933' : '#b85c00'};
    `
    const commonClass = css`
        width: 100%;
        padding-bottom: 100%;
        position: relative;
    `
    const pos = `${x}:${y}`
    return (
        <div data-pos={pos} className={[commonClass, uniqueClass].join(' ')}>
            {children}
        </div>
    )
}
