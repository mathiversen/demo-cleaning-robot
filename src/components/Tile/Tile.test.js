import React from 'react'
import Tile from './Tile'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
    const mockProps = {
        x: 5,
        y: 5,
        clean: true,
        children: [<div key="x">child</div>]
    }
    const tree = renderer.create(<Tile {...mockProps} />).toJSON()
    expect(tree).toMatchSnapshot()
})
