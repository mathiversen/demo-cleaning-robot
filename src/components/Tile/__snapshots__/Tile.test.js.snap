// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="go2471620104 go713974525"
  data-pos="5:5"
>
  <div>
    child
  </div>
</div>
`;
